src/polymer/root-selector-to-html.ts

```diff
@@ -123,14 +123,20 @@ class RootSelectorToHtml extends Rule {
         const sourceRange = containingDoc.sourceRangeForShadyRange(
             {start, end: start + match[0].length});
 
+        // Only fix plain `:root` selectors
+        let fix: Replacement[] = [];
```

Comment:
I think this should be instead:

```typescript
let fix: undefined|Replacement[] = undefined;
if (node.selector.match(/^:root$/) !== null) {
  fix = [{range: sourceRange, replacementText}];
}
```

Otherwise I think we tell the user that the warning could be fixed / was fixed even when we can't.

Reply:
Okay, for some reason, I thought I saw something like this elsewhere. I will make that change.